components/browser/errorpages/build.gradle

```diff
@@ -35,6 +35,9 @@ dependencies {
     implementation Deps.support_annotations
 
     implementation Deps.kotlin_stdlib
+
+    compileOnly "org.mozilla.geckoview:geckoview-beta-armeabi-v7a:${rootProject.ext.geckoBeta['version']}"
```

Comment:
We want our components to stay as composable as possible, so including a specific engine as dependency for the error pages will be a problem.

Another approach would be to implement the mapping for each engine (gecko-*, system-engine) in their own components, and having them accessible via the same interface in `concept-engine`.

We could then share a unified `ErrorType` across system and gecko engines, keeping the string resources in the `errorpages` component.

To be more concrete, the steps would follow as such:
1. In `concept-engine` there would be `Engine#errorTypeFromCode`.
2. `engine-gecko-*` would implement this to provide the mapping from this patch's `ErrorPages#mapGeckoViewErrorCodeToGeckoErrorType`
3. `engine-system` would implement the `ErrorPages#mapWebViewErrorCodeToErrorType`
4. In `errorpages` we would take that ErrorType and produce the string resource
5. The other helper methods can stay in this component, but might need to be refactored to work with the above.

Reply:
Sorry I'm not fully familiar with the components architecture. Made the refactor, hope that looks better. Thanks for the thorough review, really appreciated.